RobotRaconteurCore/CMakeLists.txt

```diff
@@ -336,12 +336,12 @@ if(UNIX AND NOT APPLE AND NOT ANDROID)
 
         target_include_directories(RobotRaconteurCore PRIVATE ${DBUS_INCLUDE_DIR} ${DBUS_INCLUDE_ARCH_DIR}
                                                               ${LIBUSB_INCLUDE_DIR})
-        target_link_libraries(RobotRaconteurCore PRIVATE OpenSSL::SSL OpenSSL::Crypto pthread rt z dl)
+        target_link_libraries(RobotRaconteurCore PRIVATE OpenSSL::SSL OpenSSL::Crypto)
         target_compile_definitions(RobotRaconteurCore PRIVATE ROBOTRACONTEUR_USE_OPENSSL)
     else()
         target_include_directories(RobotRaconteurCore PRIVATE ${DBUS_INCLUDE_DIR} ${DBUS_INCLUDE_ARCH_DIR}
                                                               ${LIBUSB_INCLUDE_DIR})
-        target_link_libraries(RobotRaconteurCore PUBLIC OpenSSL::SSL OpenSSL::Crypto pthread rt z dl)
+        target_link_libraries(RobotRaconteurCore PUBLIC OpenSSL::SSL OpenSSL::Crypto)
```

Comment:
As an aside, controlling PUBLIC vs. PRIVATE in response to BUILD_SHARED_LIBS seems ... questionable, as the whole point of PUBLIC vs. PRIVATE is to handle discrepancies like that, but I suggested the smaller edit.

Reply:
When using the header only libraries in boost it is necessary to have some of the libraries as public dependencies.

Reply:
> When using the header only libraries in boost it is necessary to have some of the libraries as public dependencies.

PRIVATE dependencies for static libs are added to the link line; CMake understands that static libs don't hide dependencies like that.